### PR TITLE
check CRC signature in sideChainPow tx

### PR DESCRIPTION
**Pull request by yz89**

### CODE REVIEW

File: spv/service.go [hunk 1]
```diff
@@ -95,12 +99,22 @@ func (s *Service) VerifyTransaction(tx *types.Transaction) error {
...
-		return errors.New("[VerifyElaHeader] Verify ela header failed.")
+func (s *Service) VerifyCRCArbiter(publicKey []byte) error {
+	isCRC := false
+	for _, v := range s.chainParams.CRCArbiters {
+		CRC, err := common.HexStringToBytes(v)
+		if err != nil {
+			return err
+		}
+		if bytes.Equal(CRC, publicKey) {
+			isCRC = true
```
Comment (alexpan0610) [inline spv/service.go]: Just `return nil` here and `return errors.New("CRC arbiter expected")` in the end

File: blockchain/validator.go [hunk 2]
```diff
@@ -272,6 +273,34 @@ func (v *Validator) checkProofOfWork(header *types.Header, powLimit *big.Int) (e
 	return nil
 }
 
+func (v *Validator) checkCRCArbiterSignature(sideChainPowTx *elatypes.Transaction) error {
```
Comment (alexpan0610) [inline blockchain/validator.go]: I mean move this whole `checkCRCArbiterSignature` method into SPV service, so `elatypes` and `elapayload` will no longer needed.

Comment (yz89) [reply blockchain/validator.go]: SPV service also needs `elatypes` and `elapayload`.

Comment (yz89) [reply blockchain/validator.go]: The validation of auxpow is independent of the SPV module

File: spv/service.go (hunk 1 above)
Comment (yz89) [reply spv/service.go]: fixed

File: blockchain/validator.go (hunk 2 above)
Comment (yz89) [reply blockchain/validator.go]: fixed

File: spv/service.go [hunk 3]
```diff
@@ -95,12 +101,44 @@ func (s *Service) VerifyTransaction(tx *types.Transaction) error {
...
+	buf := new(bytes.Buffer)
+	if err := payload.SerializeUnsigned(buf, elapayload.SideChainPowVersion); err != nil {
+		return err
+	}
+	pubKey, err := crypto.DecodePoint(publicKey)
 	if err != nil {
-		return errors.New("[VerifyElaHeader] Verify ela header failed.")
+		return err
+	}
+	if err := crypto.Verify(*pubKey, buf.Bytes(), payload.Signature); err != nil {
```
Comment (alexpan0610) [inline spv/service.go]: Just `return crypto.Verify(*pubKey, buf.Bytes(), payload.Signature)`

Comment (yz89) [reply spv/service.go]: fixed

**Pull request merged**